src/core/public/chrome/utils.ts

```diff
@@ -173,3 +188,28 @@ export function getOrderedLinksOrCategories(
 
   return result.sort(sortBy('order'));
 }
+
+export const getSortedNavLinks = (
+  navLinks: ChromeNavLink[],
+  enricher?: (currentItem: LinkItem, parentItem?: LinkItem) => LinkItem
+) => {
+  const sortedNavLinksTree = getOrderedLinksOrCategories(navLinks);
+  const acc: ChromeNavLink[] = [];
+  walkLinkItemsTree(
+    {
+      linkItems: sortedNavLinksTree,
+    },
+    (props) => {
+      const { currentItem, parentItem } = props;
+      const enricheredResult = enricher ? enricher(currentItem, parentItem) : currentItem;
+      if (enricheredResult.itemType === LinkItemType.LINK) {
+        acc.push(enricheredResult.link);
+      } else if (enricheredResult.itemType === LinkItemType.PARENT_LINK) {
+        if (enricheredResult.link) {
+          acc.push(enricheredResult.link);
+        }
+      }
```

Comment:
Can we simply do this? I'm curious why it doesn't check `enricheredResult.link` for `LinkItemType.LINK`
```suggestion
      if (enricheredResult.itemType === LinkItemType.LINK || enricheredResult.itemType === LinkItemType.PARENT_LINK) {
        if (enricheredResult.link) {
          acc.push(enricheredResult.link);
        }
      }
```

Reply:
Sure, updated.